packages/core/src/lib/base_protocol.ts

```diff
@@ -50,6 +50,50 @@ export class BaseProtocol implements IBaseProtocol {
     );
     return peer;
   }
+
+  /**
+   * Retrieves a list of peers based on the specified criteria.
+   *
+   * @param numPeers - The number of peers to retrieve. If you need all available peers, specify a large number.
+   * @param includeBootstrap - If true, includes a bootstrap peer in the result. Useful for protocols like Filter and Store that require only one peer for now.
+   * @returns A Promise that resolves to an array of peers based on the specified criteria.
+   */
+  protected async getPeers(
+    numPeers: number,
+    includeBootstrap: boolean
+  ): Promise<Peer[]> {
+    // Retrieve all peers that support the protocol
+    const allPeersForProtocol = await getPeersForProtocol(this.peerStore, [
+      this.multicodec
+    ]);
+
+    // Filter the bootstrap peer if required to include
+    const bootstrapPeer = includeBootstrap
+      ? allPeersForProtocol.find((peer) => peer.tags.has(Tags.BOOTSTRAP))
+      : undefined;
+
+    // Filter remaining peers excluding bootstrap
+    const remainingPeers = allPeersForProtocol.filter(
+      (peer) => peer !== bootstrapPeer
+    );
+
+    // Initialize the list of selected peers
+    const selectedPeers: Peer[] = [];
+
+    // Add the bootstrap peer if available and required
+    if (bootstrapPeer) {
+      selectedPeers.push(bootstrapPeer);
+    }
+
+    // Fill up to numPeers with remaining random peers if needed
+    while (selectedPeers.length < numPeers && remainingPeers.length > 0) {
+      selectedPeers.push(remainingPeers.shift()!);
+    }
+
+    // Trim the result to the specified number of peers if more were added
```

Comment:
why trimming? the `while` loop jsut above ensured that you have the right number.